File: token/ctoken/program/src/state.rs
use borsh::{BorshDeserialize, BorshSerialize};
use solana_program::pubkey::Pubkey;

/// Crosschain global config
#[derive(Clone, Debug, BorshDeserialize, BorshSerialize)]
pub struct Config {
    /// Initialized state.
    pub is_initialized: bool,

    /// Owner
    pub owner: Pubkey,

    /// Authority for cToken
    pub authority: Pubkey,

    /// Bridge fee
    pub fee: u64,

    /// Fee collector
    pub fee_collector: Pubkey,
}

/// Crosschain Token
#[derive(Clone, Debug, BorshDeserialize, BorshSerialize)]
pub struct CToken {
    /// Initialized state.
    pub is_initialized: bool,

    /// Bump seed used in program address.
    /// The program address is created deterministically with the bump seed,
    /// cToken program id, and cToken account pubkey. This program address has
    /// authority over the cToken's token account and token mint.
    pub bump_seed: u8,

    /// Program ID of the tokens being exchanged.
    pub token_program_id: Pubkey,

    /// CToken config
    pub config: Pubkey,

    /// Token account for cToken bridge
    pub token: Pubkey,
    /// Mint information for token
    pub token_mint: Pubkey,

    /// Bridge chain id
    /// 0: solana
    /// 4689: IoTeX
    pub destination: u32,

    /// Index for bridge instruction
    pub index: u64,

    /// Max amount for bridge
    pub max: u64,

    /// Min amount for bridge
    pub min: u64,
}

File: token/ctoken/js/test/state.ts
import {PublicKey, Connection, clusterApiUrl} from '@solana/web3.js';
import * as borsh from 'borsh';
import {cTokenAccount, cTokenAccountSchema} from '../src';

async function main() {
    // const rpc = clusterApiUrl('devnet');
    const rpc = `${process.env.SOLANA_RPC_URL}`;
    const connection = new Connection(rpc, 'confirmed');

    const cToken = new PublicKey(`${process.env.C_TOKEN}`);

    // const [authority, _bumpSeed] = PublicKey.findProgramAddressSync(
    //     [cToken.toBuffer()],
    //     cTokenProgramId,
    // );

    const cTokenData = await connection.getAccountInfo(cToken, 'confirmed');
    const cTokenAccountState = borsh.deserialize(
        cTokenAccountSchema,
        cTokenAccount,
        cTokenData!.data,
    );
    // @ts-ignore
    const is_initialized = cTokenAccountState.initialized;
    // @ts-ignore
    const bumpSeed = cTokenAccountState.bump_seed;
    // @ts-ignore
    const tokenProgramId = new PublicKey(cTokenAccountState.token_program_id);
    // @ts-ignore
    const config = new PublicKey(cTokenAccountState.config);
    // @ts-ignore
    const token = new PublicKey(cTokenAccountState.token);
    // @ts-ignore
    const tokenMint = new PublicKey(cTokenAccountState.token_mint);
    // @ts-ignore
    const destination = cTokenAccountState.destination;
    // @ts-ignore
    const index = cTokenAccountState.index;
    // @ts-ignore
    const max = cTokenAccountState.max;
    // @ts-ignore
    const min = cTokenAccountState.min;

    console.log(`cToken ${process.env.C_TOKEN}:
    {
        is_initialized: ${is_initialized},
        bump_seed: ${bumpSeed},
        token_program_id: ${tokenProgramId},
        config: ${config},
        token: ${token},
        token_mint: ${tokenMint},
        destination: ${destination},
        index: ${index},
        max: ${max},
        min: ${min}
    }`);
}

main();
